src/Runner.Worker/Handlers/HandlerFactory.cs

```diff
@@ -55,7 +55,20 @@ public sealed class HandlerFactory : RunnerService, IHandlerFactory
             else if (data.ExecutionType == ActionExecutionType.NodeJS)
             {
                 handler = HostContext.CreateService<INodeScriptActionHandler>();
-                (handler as INodeScriptActionHandler).Data = data as NodeJSActionExecutionData;
+                var nodeData = data as NodeJSActionExecutionData;
+                
+                // With node12 EoL in 03/2022, we want to be able to uniformly upgrade all JS actions to node16 from the server
+                if (executionContext.Global.Variables.GetBoolean("DistributedTask.ForceGithubJavascriptActionsToNode16") ?? false)
+                {   
+                    // The user can opt out of this behaviour by setting this variable to true, either in 'env' in their workflow or as an environment variable on their machine
+                    executionContext.Global.EnvironmentVariables.TryGetValue(Constants.Variables.Actions.AllowActionsUseUnsecureNodeVersion, out var optOutWorkflowEnv);                    
+                    var optOutMachineEnv = Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowActionsUseUnsecureNodeVersion);
+                    if (!StringUtil.ConvertToBoolean(optOutWorkflowEnv ?? optOutMachineEnv))
```

Comment:
we should convert these to bool individually and check if either is true imo.

WE could have a case here where  optOutWorkflowEnv is specifically set to "false" so we ignore the machineEnv. I don't envision it being likely, but may be better to cover the scenario anyways.

Reply:
So, my logic here was to see if there's a `optOutWorkflowEnv` set, then evaluate it. If not, fall back to machine env vars and do the same.

> WE could have a case here where optOutWorkflowEnv is specifically set to "false" so we ignore the machineEnv.

Doesn't that happen here, though? If `optOutWorkflowEnv` is `"false"`, then `optOutMachineEnv` would be ignored